app/models/service_plan.rb

```diff
@@ -10,4 +11,45 @@ class ServicePlan < ApplicationRecord
   has_many   :service_instances
 
   acts_as_taggable
+
+  def order(catalog_id, additional_parameters)
+    payload = external_payload(catalog_id, additional_parameters)
+
+    response = service_catalog_client.order_service_plan(payload)
+
+    parsed_data = JSON.parse(response.body)
```

Comment:
Could you parse the response.body and return the hash?  That would simplify this a good bit.